File: Katas/JavaScript/8KYU/I love you, a little , a lot, passionately ... not at all.js
function howMuchILoveYou(nbPetals) {
    while (nbPetals - 6 > 0)
        nbPetals -= 6

    switch (nbPetals) {
        case 1:
            nbPetals == 1
            return "I love you"
            break;
        case 2:
            nbPetals == 2
            return "a little"
            break;
        case 3:
            nbPetals == 1
            return "a lot"
            break;
        case 4:
            nbPetals == 2
            return "passionately"
            break;
        case 5:
            nbPetals == 1
            return "madly"
            break;
        case 6:
            nbPetals == 2
            return "not at all"
            break;
    }
}

//Other Solutions:

//const phrases = [
//    'I love you',
//    'a little',
//    'a lot',
//    'passionately',
//    'madly',
//    'not at all',
//]

//function howMuchILoveYou(n) {
//    return phrases[(n - 1) % phrases.length]
//}

//howMuchILoveYou = n => ['I love you', 'a little', 'a lot', 'passionately', 'madly', 'not at all'][(n - 1) % 6]

File: Katas/C#/8KYU/I love you, a little , a lot, passionately ... not at all.cs
using System;

public class Kata
{
    public static string HowMuchILoveYou(int nb_petals)
    {
        string[] phrases = new string[6] { "I love you", "a little", "a lot", "passionately", "madly", "not at all" };

        return phrases[(nb_petals - 1) % phrases.Length];
    }
}

//Other Solutions:

//using System;

//public class Kata
//{
//    public static string HowMuchILoveYou(int nb_petals)
//    {
//        return new string[] { "I love you", "a little", "a lot", "passionately", "madly", "not at all" }[(nb_petals - 1) % 6];
//    }
//}

//using System;

//public class Kata
//{
//    public static string HowMuchILoveYou(int nb_petals)
//    {
//        switch (nb_petals % 6)
//        {
//            case 5:
//                return "madly";
//            case 1:
//                return "I love you";
//            case 2:
//                return "a little";
//            case 3:
//                return "a lot";
//            case 4:
//                return "passionately";
//            case 0:
//                return "not at all";
//        }
//        return "null";
//    }
//}